sunpy/io/fits.py

```diff
@@ -71,7 +71,8 @@ def read(filepath, hdus=None, memmap=None, **kwargs):
             elif isinstance(hdus, collections.Iterable):
                 hdulist = [hdulist[i] for i in hdus]
 
-        hdulist.verify('silentfix+warn')
+        hdulist = fits.hdu.HDUList(hdulist)
+        [h.verify('silentfix+warn') for h in hdulist]
```

Comment:
Is this acceptable coding style compared to an explicit loop or to a `map()` call?  It's a list comprehension to create a list that isn't actually used...  The answer can be "yes"; I'm simply asking.

Reply:
From the [standard library docs](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions):

> List comprehensions provide a concise way to create lists.

So no, this should probably just be a `for` loop.

Reply:
Fixed!